lib/compiler/src/compiler.rs

```diff
@@ -63,47 +63,240 @@ where
-        wasm_features.set(WasmFeatures::COMPONENT_MODEL_NESTED_NAMES, false);
-
-        let mut validator = Validator::new_with_features(wasm_features);
-        validator
-            .validate_all(data)
-            .map_err(|e| CompileError::Validate(format!("{e}")))?;
+
+        // Parse the module and check for feature usage
+        for payload in Parser::new(0).parse_all(data) {
+            match payload {
+                Ok(Payload::TagSection(_)) => {
+                    // The presence of a tag section means exceptions are used
+                    detected_features.exceptions = true;
+                }
+                Ok(Payload::ImportSection(reader)) => {
+                    // Check for imported tags (exceptions)
+                    for import in reader.into_iter().flatten() {
+                        match import.ty {
+                            wasmparser::TypeRef::Tag(_) => {
+                                detected_features.exceptions = true;
+                            }
+                            wasmparser::TypeRef::Memory(memory_type) => {
+                                if memory_type.memory64 {
+                                    detected_features.memory64 = true;
+                                }
+                                if memory_type.shared {
+                                    detected_features.threads = true;
+                                }
+                            }
+                            _ => {}
+                        }
+                    }
+                }
+                Ok(Payload::ExportSection(reader)) => {
+                    // Check for exported tags (exceptions)
+                    for Export { kind, .. } in reader.into_iter().flatten() {
+                        if kind == wasmparser::ExternalKind::Tag {
+                            detected_features.exceptions = true;
+                        }
+                    }
+                }
+                Ok(Payload::CodeSectionEntry(reader)) => {
+                    // Analyze code section for feature usage without parsing individual operators
+                    if let Ok(_op_reader) = reader.get_operators_reader() {
+                        // Get the raw binary data of the function body
+                        let raw_data = reader.range();
+                        let data = &data[raw_data.start..raw_data.end];
+
+                        // Since we can't easily iterate through operators, we'll look for
+                        // characteristic byte sequences in the raw data
+
+                        // Check for SIMD operations (0xfd prefix)
+                        if data.contains(&0xfd) {
+                            detected_features.simd = true;
+                        }
+
+                        // Check for exception handling opcodes (0x06, 0x07, 0x08)
+                        if data.contains(&0x06) || data.contains(&0x07) || data.contains(&0x08) {
```

Comment:
This is being picked up by modules that should not contain eh-related opcodes; for example: 
```
λ target/release/wasmer run cowsay --entrypoint=cowsay -- "Hello"
error: Spawn failed
╰─▶ 1: compile error: Validate("This module uses the exceptions feature, which requires the LLVM backend. Use --enable-exceptions flag or switch to the LLVM backend.")
```